search/includes/classes/class-versioningcleanupjob.php

```diff
@@ -34,7 +34,8 @@ public function init() {
 	 */
 	public function schedule_job() {
 		if ( ! wp_next_scheduled( self::CRON_EVENT_NAME ) ) {
-			wp_schedule_event( time(), 'weekly', self::CRON_EVENT_NAME );
+			// phpcs:disable WordPress.WP.AlternativeFunctions.rand_mt_rand
+			wp_schedule_event( time() + ( mt_rand( 1, 60 ) * HOUR_IN_SECONDS ), 'weekly', self::CRON_EVENT_NAME );
```

Comment:
Staggering potentially thousands of sites between 1 and 60 isn't really all that much stagger tbh, many will still overlap. Since we're dealing with a "weekly" thing, I'd vote to stagger between however many minutes exist in a week: `1 to (WEEK_IN_SECONDS / MINUTE_IN_SECONDS)`

The downside being that it could take up to a week to run for the first time, but thats probably okay for this sort of task? If it is a concern, it could be combatted by scheduling a one-time immediate job alongside the recurring job - but I think it's probably overkill since the cleanup can wait.

Reply:
Good point. I just wanted to stagger though, so they didn't all end up at the same second (within a few seconds of each other). But it does make sense to extend the time window.